src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2742,192 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
+        
+    return vmd
+
+
+def _convert_tescan_to_odemis_metadata(metadata: bytes) -> dict:
+    """
+    Converts TESCAN metadata to odemis metadata format
+    :param metadata: metadata bytes (TESCAN format)
+    :return: (dict) metadata as dictionary (odemis format)
+    :raises: (ValueError): if the metadata cannot be parsed
+    """
+
+    # parse metadata bytes as str (valid metadata)
+    try:    
+        vmd: str = _parse_tescan_metadata(metadata)
+    except ValueError as e:
+        raise ValueError(f"Failed to parse TESCAN metadata: {e}")
+
+    # parse the metadata string as ini
+    # NOTE: ConfigParser converts keys to case-insensitive (lower) by default
+    # so metadata keys are all lower case (this behaviour can be altered, see link below)
+    # https://stackoverflow.com/questions/19359556/configparser-reads-capital-keys-and-make-them-lower-case
+    tmd = configparser.ConfigParser()
+    tmd.read_string(vmd)
+        
+    md = {}
+
+    # general
+    try:
+        datetime_str = tmd["MAIN"]["date"] + " " + tmd["MAIN"]["time"]
+        date = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_ACQ_DATE} - {type(e)}: {e}")
+
+    try:
+        md[model.MD_HW_NAME] = tmd["MAIN"]["device"] + "-" + tmd["MAIN"]["serialnumber"]    
+    except KeyError as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {type(e)}: {e}")
+    try:
+        md[model.MD_HW_VERSION] = tmd["MAIN"]["device"] + "-" + tmd["MAIN"]["devicemodel"]    
+    except KeyError as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {type(e)}: {e}")
+    
+    try:
+        md[model.MD_SW_VERSION] = tmd["MAIN"]["softwareversion"]
+    except KeyError as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_SW_VERSION} - {type(e)}: {e}")
+
+    try:
+        md[model.MD_PIXEL_SIZE] = (float(tmd["MAIN"]["pixelsizex"]), 
+                                   float(tmd["MAIN"]["pixelsizey"]))
+    except KeyError as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_PIXEL_SIZE} - {type(e)}: {e}")
+
+    # beam metadata
+
+    try:
+        # beam types is the section header of SEM/FIB metadata
+        beam_type = "SEM" if "SEM" in tmd.sections() else "FIB"
+        
+        # check if beam type is valid
+        assert beam_type in tmd.sections()
+            
+        try:
+            md[model.MD_DWELL_TIME] = float(tmd[beam_type]["dwelltime"])
+        except KeyError as e:
+            logging.warning(f"Failed to parse TESCAN metadata: {model.MD_DWELL_TIME} - {e}")
+        try:
+            # MD_POS should match image dimensions, e.g. only XY for 2D, XYZ for 3D, etc.
+            md[model.MD_POS] = (float(tmd[beam_type]["stagex"]), 
+                            float(tmd[beam_type]["stagey"]),
```

Comment:
Please adjust the indentation. Typically you want the two elements (`float`...) to be aligned.